flight/UAVObjects/uavobjectmanager.c

```diff
@@ -377,7 +397,7 @@ UAVObjHandle UAVObjRegister(uint32_t id,
  */
 UAVObjHandle UAVObjGetByID(uint32_t id)
 {
-	UAVObjHandle * found_obj = (UAVObjHandle *) NULL;
+	UAVObjHandle found_obj = NULL;
```

Comment:
IMO it is preferable to what we had before, where it was a uintptr_t or void * that one always casts back and forth. 

I think we need to move to these practices:

* In public modules, always use anonymous structure pointers for handles instead of uintptr_t or void *
* Fully define these structures in private headers for the module that needs to see the information.
* Do not include private headers everywhere like we currently do.  Preferably place them somewhere so that include paths don't work out for other modules to include them.

Otherwise, there's basically no type safety (because anything can roundtrip through a void * or cast)  and there's basically no encapsulation (since everyone can see the private header informaiton).